app/models/miq_report/seeding.rb

```diff
@@ -84,12 +84,32 @@ def seed_record(path, report)
       end
     end
 
+    def template_type(path)
+      File.dirname(path).include?("report") ? "report" : "compare"
+    end
+
     def seed_files
-      Dir.glob(REPORT_DIR.join("**/*.yaml")).sort + Dir.glob(COMPARE_DIR.join("**/*.yaml")).sort
+      seed_core_files + seed_plugin_files
+    end
+
+    def seed_core_files
+      [REPORT_DIR, COMPARE_DIR].flat_map { |dir| Dir.glob(dir.join("**", "*.yaml")).sort }
+    end
+
+    def seed_plugin_files
+      Vmdb::Plugins.flat_map do |plugin|
+        %w[reports compare].flat_map { |dir| Dir.glob(plugin.root.join("content", dir, "**", "*.yaml")).sort }
+      end
     end
 
     def seed_filename(path)
-      path.remove("#{REPORT_DIR}/").remove("#{COMPARE_DIR}/")
+      if File.dirname(path).include?("reports")
+        path.split("reports/").last
+      elsif File.dirname(path).include?("compare")
+        path.split("compare/").last
+      else
+        path
```

Comment:
I realized that `seed_filename` is sometimes passed not the full path but just the filename from the db record [here](https://github.com/ManageIQ/manageiq/blob/6aabf1967e0c5dc3d2396806516a5d6499666916/app/models/miq_report/seeding.rb#L11) and [here](https://github.com/ManageIQ/manageiq/blob/6aabf1967e0c5dc3d2396806516a5d6499666916/app/models/miq_report/seeding.rb#L20), so it is possible to not have either "reports" or "compare" in the path.

Since those records have the filename set by `seed_filename(path)` I don't think we could `index_by(&:filename)` which simplifies this method but want to check to see if I'm missing something.